strapi/strapi
# Typegen support for the develop command

### What does it do?

- It makes automatic type generation parts of the `develop` command core behavior. It cannot be disabled.
- Always clean up the "generated" types' folder before starting a type gen process
- It makes sure the "generated" types folder is removed before building the user project during the `develop` reload 

### Why is it needed?

With the introduction of the new Strapi's type system, we're improving the TS integration in our toolchain in order to provide a better developer experience. The automatic type generation is part of this & this PR aims to both fix existing issues related to generated types preventing the app to run and enable the automatic type generation for all users

Additional context: https://github.com/strapi/strapi/pull/17349#issuecomment-1639921212

### How to test it?

- `cd examples/kitchensink-ts`
- `yarn strapi ts:generate-types`
- `yarn develop`
- Create a new content type
- The server should reload successfully, and the types located in `types/generated/content-types.d.ts` should have been updated
- Try the same process with a javascript example app such as `examples/getstarted`

### Related issue(s)/PR(s)

fix [#16560 ](https://github.com/strapi/strapi/issues/17084)
